app/controllers/data_requests_controller.rb

```diff
@@ -71,6 +75,10 @@ def exportable_type
     end
   end
 
+  def verify_scope
+    head 404 unless unscoped_exportable
```

Comment:
minor - not blocking. I found the `unscoped_*` terminology strange and found it hard to follow. Aren't these the data request parent resources, i.e. `workflow`, `project` and `exportable`, perhaps `exportable_*` is a better prefix.

Reply:
I'm confused as to why are there multiple ways of finding the workflow / project exportable resources, i.e. https://github.com/zooniverse/caesar/blob/97832c797551d90848468e880577bf2a61c27d81/app/controllers/data_requests_controller.rb#L74
vs https://github.com/zooniverse/caesar/blob/97832c797551d90848468e880577bf2a61c27d81/app/controllers/data_requests_controller.rb#L86

Don't we want to ensure user roles are tested for authorization to access the resources ? It should 404 if i don't have any authorized access instead of looking up if the resource exists generally via id. 

To my mind the public resources should be handled by the `policy_scope` lookup but they look like they aren't https://github.com/zooniverse/caesar/blob/master/app/policies/project_policy.rb so this is probably why this skip authorization lookup has crept in. Thoughts?

Reply:
we have flip-flopped a few times on whether public data exports should be accessible to people who don't have access to the workflow or project for which they were generated, which is probably why there's cruft here. i don't know what the desired behavior is at this point.

Reply:
This may also be related to whether or not a request to create a data request must come from an authenticated user, because I'm not sure that the i2a app is.

Reply:
The public export must truly be public for the exports to work for the I2A classrooms use case. The instructors won't be owners or collabs on the project.

Reply:
Thank you @srallen that's the missing piece of the puzzle.

Reply:
It looks messy because the UI should be able to handle the normal behavior--nobody can see workflows or projects that they don't have privileges for--so when returning HTML we enforce normal scoping rules.

Reply:
> It looks messy because the UI should be able to handle the normal behavior--nobody can see workflows or projects that they don't have privileges for--so when returning HTML we enforce normal scoping rules.

I think this is the most confusing aspect for me as I assume that the `data_request_policy` scope lookups could handle finding the marked 'public' resources vs the manual setup in the controller here. 

It looks like the policy is setup to look for publicly available data as well, https://github.com/zooniverse/caesar/blob/master/app/policies/data_request_policy.rb

Reply:
The DataRequestPolicy does handle the scoping for the DataRequest objects. However, since this is a nested resource, I need to also respond with the appropriate workflow/project resource, which policy we elected not to modify.